ui/src/shared/components/FluxMonacoEditor.tsx

```diff
@@ -28,18 +34,39 @@ interface Props {
   onCursorChange?: (position: Position) => void
 }
 
-const FluxEditorMonaco: FC<Props> = props => {
+const FluxEditorMonaco: FC<Props> = ({
+  script,
+  onChangeScript,
+  onSubmitScript,
+  onCursorChange,
+}) => {
+  let completionProvider = {dispose: () => {}}
+  const lspServer = useRef(new Server(false))
+  const [docVersion, setdocVersion] = useState(2)
+  const [msgID, setmsgID] = useState(3)
+
+  useEffect(() => {
+    sendMessage(initialize, lspServer.current)
+    sendMessage(didOpen(script), lspServer.current)
+  }, [])
+
+  useEffect(() => {
+    return function cleanup() {
+      completionProvider.dispose()
+    }
+  }, [])
+
   const editorWillMount = (monaco: MonacoType) => {
+    monaco.languages.register({id: FLUXLANGID})
```

Comment:
what happens if you call this a bunch of times, does it reset any of the registrations that happen below?

Reply:
I think it might but `editorWillMount` only runs once?